src/XMakeBuildEngine/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -1144,7 +1144,7 @@ private void AddMetaprojectTargetForManagedProject(ProjectInstance traversalProj
 
             ProjectTargetInstance target = metaprojectInstance.AddTarget(targetName ?? "Build", String.Empty, String.Empty, outputItemAsItem, null, String.Empty, String.Empty, false /* legacy target returns behaviour */);
 
-            AddReferencesBuildTask(metaprojectInstance, target, targetName, outputItem);
+            AddReferencesBuildTask(metaprojectInstance, target, targetName, null);
```

Comment:
Can you add a comment above the line saying "Passing null to avoid cumulative recursive build up of outputs. See bug #396"